src/Elastic.Apm/Helpers/RequestBodyStreamHelper.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Text;
+
+namespace Elastic.Apm.Helpers
+{
+	internal static class RequestBodyStreamHelper
+	{
+		internal const int RequestBodyMaxLength = 10000;
+
+		internal static string ToString(Stream requestBody, out bool longerThanMaxLength)
+		{
+			longerThanMaxLength = false;
+
+			if (requestBody == null) return null;
+
+			string body = null;
+
+			var capacity = 512;
+			var totalRead = 0;
+			var arrayPool = ArrayPool<char>.Shared;
+			char[] buffer;
+
+			if (requestBody.CanSeek)
```

Comment:
@gregkalapos I made a mistake here when refactoring this code.
The actual idea was to do the first `StreadReader` part (for finding out the length) only in case that `requestBody.Length == 0`, otherwise we can use this value directly.

The `stream.CanSeek` parts are actually a relict and will be removed.